multiversx_sdk_core/transaction_builders/delegation_factory.py

```diff
@@ -0,0 +1,430 @@
+    DELEGATION_MANAGER_OPS = 50000000
+    DELEGATION_OPS = 1000000
+    UNSTAKE_TOKENS = 5000000
+    UNBOND_TOKENS = 5000000
+
+
+ADDITIONAL_GAS_LIMIT_PER_NODE = 6000000
+ADDITIONAL_GAS_FOR_OPERATIONS = 10000000
+
+
+class DelegationFactory:
+    def __init__(self, config: IConfig) -> None:
+        self.config = config
+
+    def create_new_delegation_contract(self,
+                                       sender: IAddress,
+                                       receiver: IAddress,
+                                       total_delegation_cap: int,
+                                       service_fee: int,
+                                       value: ITransactionValue,
+                                       transaction_nonce: Optional[INonce] = None,
+                                       gas_price: Optional[IGasPrice] = None,
+                                       gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = self._prepare_data_for_create_new_delegation_contract(total_delegation_cap, service_fee)
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=receiver,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_MANAGER_OPS + ADDITIONAL_GAS_FOR_OPERATIONS,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+
+        return transaction
+
+    def _prepare_data_for_create_new_delegation_contract(self, total_delegation_cap: int, service_fee: int) -> List[str]:
+        function = "createNewDelegationContract"
+        args_list: List[Any] = [
+            function,
+            arg_to_string(total_delegation_cap),
+            arg_to_string(service_fee)
+        ]
+        return args_list
+
+    def add_nodes(self,
+                  sender: IAddress,
+                  delegation_contract: IAddress,
+                  public_keys: Sequence[IValidatorPublicKey],
+                  signed_messages: List[ISignature],
+                  value: ITransactionValue,
+                  transaction_nonce: Optional[INonce] = None,
+                  gas_price: Optional[IGasPrice] = None,
+                  gas_limit: Optional[IGasLimit] = None) -> Transaction:
+        data = self._prepare_data_for_add_nodes(public_keys, signed_messages)
+        num_nodes = len(public_keys)
+
+        transaction = self.create_transaction(
+            sender=sender,
+            receiver=delegation_contract,
+            data_parts=data,
+            execution_gas_limit=MetaChainSystemSCsCost.DELEGATION_OPS + num_nodes * ADDITIONAL_GAS_LIMIT_PER_NODE,
+            gas_limit_hint=gas_limit,
+            gas_price=gas_price,
+            nonce=transaction_nonce,
+            value=value
+        )
+        return transaction
+
+    def _prepare_data_for_add_nodes(self, public_keys: Sequence[IValidatorPublicKey], signed_messages: List[ISignature]) -> List[str]:
```

Comment:
As above - the private function can be merged into `add_nodes` (if not reused / called by another "factory function") - for "inner" design consistency / symmetry. Having this symmetry with JS factories and other PY factories, it should be a bit easier to update the factories at the same time, in the future, when needed (e.g. new functions).

Reply:
Done!